### DISCUSSION TITLE
Redirect URLs including info similar to cURL

### DISCUSSION THREAD

**Discussion by RobertEcker** [spatie/crawler#362] [Q&A] [answered]
Hi (it's a question not issue ;-) ),

is there a way, starting from a given URL, ...

... to see all redirected URLs (similar to cURL)?
... with info such as HTTP code, total time etc. (similar to cURL, see https://www.php.net/manual/en/function.curl-getinfo.php)?

hint: javascript is needed because some pages work only with it

```
$url = 'http://example.com';
Crawler::create()
->setCrawlObserver(new PageCrawlObserver)
->setUserAgent('My browser')
->setConcurrency(1)
->setMaximumCrawlCount(99)
->setMaximumDepth(99)
->executeJavaScript()
->startCrawling($url);
```

Thank you,

Robert

**Comment by Redominus** [comment, marked as answer]
Hi!
Read the guzzle documentation about client config array. You must pass that array to Crawler::create($configArray).
The crawler observer will receive a Response interface from guzzle, again read the guzzle documentation about getting redirect history from responses.
cheers
